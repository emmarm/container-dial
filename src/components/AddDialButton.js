import React from 'react';
import PropTypes from 'prop-types';
import styled from 'react-emotion';

import Button from './Button';

const AddButton = styled(Button)({
  background: 'rgba(255, 255, 255, 0.3)',
  borderRadius: 0,
  height: '100%',
  ':hover': {
    background: 'rgba(255, 255, 255, 0.5)'
  }
});

const AddDialButton = ({ handleShowDialModal }) => (
  <AddButton onClick={handleShowDialModal} text="Add New" icon="add" />
);

AddDialButton.propTypes = {
  handleShowDialModal: PropTypes.func.isRequired
};

export default AddDialButton;
