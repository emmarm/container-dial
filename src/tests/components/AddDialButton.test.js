import React from 'react';
import { shallow } from 'enzyme';

import AddDialButton from '../../components/AddDialButton';

describe('AddDialButton component', () => {
  const props = {
    handleShowDialModal: jest.fn()
  };
  const wrapper = shallow(<AddDialButton {...props} />);

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('calls handleShowDialModal on click', () => {
    const { handleShowDialModal } = props;
    const button = wrapper.find('Styled(button)');
    expect(button.length).toBe(1);
    button.simulate('click');
    expect(handleShowDialModal).toHaveBeenCalled();
  });
});
