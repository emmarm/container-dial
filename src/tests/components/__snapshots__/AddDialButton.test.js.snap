// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddDialButton component matches snapshot 1`] = `
<Styled(Button)
  icon="add"
  onClick={[Function]}
  tabIndex={1}
  text="Add New"
/>
`;
